editor/editor_undo_redo_manager.cpp

```diff
@@ -248,42 +249,16 @@ void EditorUndoRedoManager::commit_action(bool p_execute) {
 		return;
 	}
 
-	if (!history.undo_stack.is_empty()) {
-		// Discard action if it should be merged (UndoRedo handles merging internally).
-		switch (pending_action.merge_mode) {
-			case UndoRedo::MERGE_DISABLE:
-				break; // Nothing to do here.
-			case UndoRedo::MERGE_ENDS: {
-				if (history.undo_stack.size() < 2) {
-					break;
-				}
-
-				const Action &prev_action = history.undo_stack.back()->get();
-				const Action &pre_prev_action = history.undo_stack.back()->prev()->get();
-				if (pending_action.merge_mode == prev_action.merge_mode && pending_action.merge_mode == pre_prev_action.merge_mode &&
-						pending_action.action_name == prev_action.action_name && pending_action.action_name == pre_prev_action.action_name) {
-					pending_action = Action();
-					is_committing = false;
-					emit_signal(SNAME("history_changed"));
-					return;
-				}
-			} break;
-			case UndoRedo::MERGE_ALL: {
-				const Action &prev_action = history.undo_stack.back()->get();
-				if (pending_action.merge_mode == prev_action.merge_mode && pending_action.action_name == prev_action.action_name) {
-					pending_action = Action();
-					is_committing = false;
-					emit_signal(SNAME("history_changed"));
-					return;
-				}
-			} break;
-		}
+	if(merging) {
+		pending_action = Action();
+		is_committing = false;
+		emit_signal(SNAME("history_changed"));
```

Comment:
Move these two shared lines out of the statement

Reply:
Will push change in a sec.

Reply:
Change made